GG/src/ListBox.cpp

```diff
@@ -686,9 +686,7 @@ void ListBox::ChildrenDraggedAway(const std::vector<Wnd*>& wnds, const Wnd* dest
 
         m_selections = new_selections;
 
-        if (m_selections.empty()) {
-            ClearedSignal();
-        } else if (m_selections.size() != initially_selected_rows.size()) {
+        if (m_selections.size() != initially_selected_rows.size()) {
```

Comment:
I think that the original intent of the ClearedSignal was to indicate that the ListBox had been cleared not that it had become empty by losing its last row.

The ClearedSignal is only ever emitted in the ListBox code when the ListBox is cleared.  It is not emitted when rows are deleted.

The ClearedSignal is only ever consumed in the QueueListBox, which is a NOSEL ListBox so it would never have triggered the removed section of code.  

However, a LIST_SEL or LIST_SINGLESEL box will trigger the ClearedSignal if its last selected row is dragged away, even though there are other rows in the list.